src/generator/helpers/constraints.ts

```diff
@@ -0,0 +1,81 @@
+import { DMMF } from '@prisma/client/runtime'
+
+/**
+ * Find the unique identifiers necessary to indentify a field
+ *
+ * Unique fields for a model can be one of (in this order):
+ * 1. One (and only one) field with an @id annotation
+ * 2. Multiple fields with @@id clause
+ * 3. One (and only one) field with a @unique annotation (if there are multiple, use the first one)
+ * 4. Multiple fields with a @@unique clause
```

Comment:
This is a great start to helpful docs around this topic.

I think even more detail and examples are warranted when it comes to the user-facing side.

I would like detailed and contextual JSDoc available on the generated resolvers.

Then also general doc in the README.

I am ok keeping the JSDoc bar low here and treating rich docs as a separate issue.

Reply:
Good point. Yes we can add JSDoc details in a future PR.